CONTRIBUTING.md

```diff
@@ -44,3 +44,20 @@ For static security check, run:
 For type annotation test, run:
 
     mypy pydax
+
+## Where to Expose a Symbol (Function, Class, etc.)?
+
+Generally speaking:
+
+- If a symbol is likely used by a casual user regularly, it should be exposed in `pydax/__init__.py`. This gives casual
+  users the cleanest and the most direct access.
+- If a symbol is used only by a power user, but is unlikely used by a casual user regularly, it should be exposed in a
+  file that does not start with an underscore, such as `pydax/schema.py`; or in the `__init__.py` file in a subdirectory
+  that does not start with an underscore, such as `pydax/loaders/__init__.py`. The rationale is that the amount of such
+  symbols is usually large and if we expose them at the root level of the package, it would be messy and likely confuse
+  casual users.
+- If a symbol is solely used for internal purpose, it should be exposed only in files starting with a single underscore,
+  such as `pydax/_dataset.py`.
+
+Please keep in mind that the criteria above are not meant to be rigid: They should be applied flexibly in light of
+factors such as where existing symbols are placed and other potentially important considerations (if any).
```

Comment:
> Does that mean in the future when I pull in symbols into new modules, I should be importing them from where they were publicly exposed as opposed to from where their source code is defined?

Can we actually add this takeaway as a bullet point into this section? Will be useful for me to remember. Maybe something like:

> - Keep in mind when referencing public symbols in new modules, you should import the symbol from where the package publicly exposes it rather than from where the source code of the symbol is defined. This way we test the user experience, e.g. use `from .schema import SchemaDict` rather than `from ._schema import SchemaDict`.

Reply:
Makes sense, I changed the wording a bit and pushed an update

Reply:
thanks looks good